src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/Formatting/FormattingHandler.ts

```diff
@@ -75,6 +115,38 @@ export class FormattingHandler {
         return this.emptyFormattingResponse;
     }
 
+    private sanitizeTextEdits(razorDocument: IRazorDocument, textEdits: vscode.TextEdit[]) {
+        const htmlDocText = razorDocument.htmlDocument.getContent();
+        const zeroBasedLineCount = this.countLines(htmlDocText);
+        const serializableTextEdits = Array<SerializableTextEdit>();
+        for (let textEdit of textEdits) {
+            // The below workaround is needed due to a bug on the HTML side where
+            // they'll sometimes send us an end position that exceeds the length
+            // of the document. Tracked by https://github.com/microsoft/vscode/issues/175298.
+            if (textEdit.range.end.line > zeroBasedLineCount ||
+                textEdit.range.start.line > zeroBasedLineCount) {
+                const lastLineLength = this.getLastLineLength(htmlDocText);
+                const updatedPosition = new vscode.Position(zeroBasedLineCount, lastLineLength);
+
+                let start = textEdit.range.start;
+                let end = textEdit.range.end;
+                if (textEdit.range.start.line > zeroBasedLineCount) {
+                    start = updatedPosition;
+                }
+
+                if (textEdit.range.end.line > zeroBasedLineCount) {
+                    end = updatedPosition;
+                }
```

Comment:
nit: new line under this one? I wonder if there's some tslint setting we can set similar to editorconfig for this